doc/development/Jenkinsfile.template

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env groovy
+
+pipeline {
+
+    agent {
+        // Use the docker to assign the Python version.
+        // Use the label to assign the node to run the test.
+        // It is recommended by SQUARE to not add the label
+        docker {
+            alwaysPull true
+            image 'lsstts/develop-env:develop'
+            args "-u root --entrypoint=''"
+        }
+    }
+
+    environment {
+        // XML report path
+        XML_REPORT="jenkinsReport/report.xml"
+        // Module name used in the pytest coverage analysis
+        MODULE_NAME="lsst.ts.{csc_name}"
+        user_ci = credentials('lsst-io')
+        LTD_USERNAME="${user_ci_USR}"
+        LTD_PASSWORD="${user_ci_PSW}"
+        work_branches = "${GIT_BRANCH} ${CHANGE_BRANCH} develop"
+    }
+
+    stages {
+        stage ('Install Requirements And Update Branches') {
+            steps {
+                // When using the docker container, we need to change
+                // the HOME path to WORKSPACE to have the authority
+                // to install the packages.
+                withEnv(["HOME=${env.WORKSPACE}"]) {
+                    sh """
+                        source /home/saluser/.setup_dev.sh || echo loading env failed. Continuing...
+                        cd /home/saluser/repos/ts_xml
+                        /home/saluser/.checkout_repo.sh ${work_branches}
+                        git pull
+                        cd /home/saluser/repos/ts_salobj
+                        /home/saluser/.checkout_repo.sh ${work_branches}
+                        git pull
+                        cd /home/saluser/repos/ts_sal
+                        /home/saluser/.checkout_repo.sh ${work_branches}
+                        git pull
+                        cd /home/saluser/repos/ts_idl
+                        /home/saluser/.checkout_repo.sh ${work_branches}
+                        git pull
+                        make_idl_files.py {csc_name}
```

Comment:
What about when a CSC needs idl files from other CSCs? Maybe add a second variable that is empty by default?